logging/logging.go

```diff
@@ -252,6 +252,69 @@ var detectedResource struct {
 	once sync.Once
 }
 
+func detectGAEResource() *mrpb.MonitoredResource {
+	return &mrpb.MonitoredResource{
+		Type: "gae_app",
+		Labels: map[string]string{
+			"project_id":  os.Getenv("GOOGLE_CLOUD_PROJECT"),
+			"module_id":   os.Getenv("GAE_SERVICE"),
+			"version_id":  os.Getenv("GAE_VERSION"),
+			"instance_id": os.Getenv("GAE_INSTANCE"),
+			"runtime":     os.Getenv("GAE_RUNTIME"),
+		},
+	}
+}
+
+// TODO: e2e test this
+func isCloudFunction() bool {
+	_, name := os.LookupEnv("FUNCTION_NAME")
+	_, target := os.LookupEnv("FUNCTION_TARGET")
+	return (name || target)
+}
+
+func isCloudRun() bool {
+	_, config := os.LookupEnv("K_CONFIGURATION")
+	return config
+}
+
+func detectGCRResource() *mrpb.MonitoredResource {
+	projectID, err := metadata.ProjectID()
+	if err != nil {
+		return nil
+	}
+	zone, err := metadata.Zone()
+	if err != nil {
+		return nil
+	}
+	return &mrpb.MonitoredResource{
+		Type: "cloud_run_revision",
+		Labels: map[string]string{
+			"project_id":         projectID,
+			"location":           zone,
+			"service_name":       os.Getenv("K_SERVICE"),
+			"revision_name":      os.Getenv("K_REVISION"),
+			"configuration_name": os.Getenv("K_CONFIGURATION"),
+		},
+	}
+}
+
+// TODO(nicolezhu): e2e test this
+func isKubernetesEngine() bool {
+	_, err := metadata.InstanceAttributeValue("cluster-name")
+	if err != nil {
+		return false
+	}
+	return true
+}
+
+// TODO(nicolezhu): e2e test this
+func detectGKEResource() *mrpb.MonitoredResource {
+	return &mrpb.MonitoredResource{
+		Type:   "gae_app",
```

Comment:
You meant gke_app ?

Reply:
Python (or maybe it's the fluent agent?) seems to be using k8s_container. It looks like gke_container is another option though